spec/index.bs

```diff
@@ -1245,6 +1238,17 @@ interface GPU {
         </div>
 </dl>
 
+<div class="example">
+    Request a {{GPUAdapter}}:
+    <pre highlight="js">
+        // using async/await
+        const adapter = await navigator.gpu.requestAdapter(/* ... */);
+
+        // using promises
+        navigator.gpu.requestAdapter(/* ... */).then(adapter => { /* ... */);
```

Comment:
I'm not sure it's particularly useful or desirable to show both the `await` syntax and the `.then()` syntax for waiting on the adapter here. In particular, I've been told in the past that effectively all new spec and explainer examples should prefer the `await` syntax to keep things as concise and readable as possible.

Interested to hear the editors opinions on that, though.

Reply:
I'll wait until we hear back from the editors, but I tend to agree with the idea of only showing the `await` syntax.

My only beef with showing only the `await` syntax is that it effectively makes async code behave like sync code. Developers tend to forget that `await` "pauses" execution until the next event loop iteration at best or for a long time when waiting for a long task. Although, developers working with WebGPU most likely will be aware of how `await` works.

**On another note:** These examples should most likely be removed once a proper example is added to the `Initialization/examples` section.

Reply:
+1 to await only, if people want to use `.then`, we can assume they understand the relationship

Reply:
updated the example to only show the `await` syntax